pyjobshop/Solution.py

```diff
@@ -8,19 +8,19 @@ class TaskData:
 
     Parameters
     ----------
-    machine
-        The assigned machine index.
+    mode
+        The selected mode.
+    machines
+        The selected machines.
     start
         The start time.
-    duration
-        The duration.
     end
         The end time.
     """
 
-    machine: int
+    mode: int
```

Comment:
To include a mode in the construction of a solution is annoying if you want to make an initial solution. Users would have to map their solution to modes which is not a straightforward thing to do.

Reply:
Excluding mode from the solution is more convenient for the user to pass initial solutions. There are downsides:
- Having multiple modes of the same resources and duration are hard to map back to.
	- But the question is why someone would have multiple identical modes? We could raise for that although I don't think that's necessary.
	- The case when there is no one-to-one mapping between modes and machines/duration is when durations are not fixed.